src/components/Breadcrumb.vue

```diff
@@ -23,44 +29,71 @@ export default {
   },
   asyncComputed: {
     breadcrumb: {
-      default: { root: {}, path: [] },
+      default: [],
       async get() {
         this.loading = true;
-        const breadcrumb = { root: {}, path: [] };
-        const { _id, _modelType, name } = this.location;
-        if (_modelType === 'folder') {
+        const breadcrumb = [];
+        // The reason for this local user variable is that
+        // we have to set up reactivity dependancy before the first async function call
+        const { user } = this.girderRest;
+        const { type, name, id } = this.location;
+        if (type === 'folder') {
           // The last breadcrumb isn't returned by rootpath.
           if (name) {
-            breadcrumb.path.unshift(this.location);
+            breadcrumb.unshift(this.location);
           } else {
-            const { data } = await this.girderRest.get(`folder/${_id}`);
-            breadcrumb.path.unshift(this.extractCrumbData(data));
+            const { data } = await this.girderRest.get(`folder/${id}`);
+            breadcrumb.unshift(this.extractCrumbData(data));
           }
           // Get the rest of the path.
-          const { data } = await this.girderRest.get(`folder/${_id}/rootpath`);
+          const { data } = await this.girderRest.get(`folder/${id}/rootpath`);
           data.reverse().forEach((crumb) => {
-            const { object } = crumb;
-            const entity = this.extractCrumbData(object);
-            if (entity._modelType === 'folder') {
-              breadcrumb.path.unshift(entity);
-            } else {
-              breadcrumb.root = entity;
-            }
+            breadcrumb.unshift(this.extractCrumbData(crumb.object));
           });
-        } else {
-          const { data } = await this.girderRest.get(`${_modelType}/${_id}`);
-          breadcrumb.root = this.extractCrumbData(data);
+        } else if (type === 'user' || type === 'collection') {
+          const { data } = await this.girderRest.get(`${type}/${id}`);
+          breadcrumb.unshift(this.extractCrumbData(data));
+        }
+        if (!this.noRoot) {
+          if (
+            type === 'users' ||
+            (user && breadcrumb.length && breadcrumb[0].type === 'user')
```

Comment:
Why is `user` part of this conditional?  If I'm not logged in and browsing a public file in some real user's space, won't this evaluate false?

Reply:
This is for the 'users' location that lists all users, recent changes of girder make it only authenticated users are able to list all users. So, if the user is not authenticated, hide the 'users' location from the breadcrumb.

Reply:
Got it, thanks.